sled-storage/src/dataset.rs

```diff
@@ -262,15 +262,25 @@ pub(crate) async fn ensure_zpool_has_datasets(
             log,
             "Ensuring encrypted filesystem: {} for epoch {}", dataset, epoch
         );
+        let name = format!("{}/{}", zpool_name, dataset);
         let result = Zfs::ensure_dataset(zfs::DatasetEnsureArgs {
-            name: &format!("{}/{}", zpool_name, dataset),
+            name: &name,
             mountpoint: Mountpoint::Path(mountpoint),
             can_mount: zfs::CanMount::On,
             zoned,
             encryption_details: Some(encryption_details),
             size_details: None,
             id: None,
             additional_options: None,
+        })
+        .map_err(|err| {
+            warn!(
+                log,
+                "Failed to ensure encrypted root filesystem";
+                "name" => ?name,
+                "err" => ?err
```

Comment:
Should this be

```suggestion
                InlineErrorChain::new(&err)
```

?